packages/blockchain-extension/extension/extension.ts

```diff
@@ -116,8 +124,30 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         const tempCommandRegistry: TemporaryCommandRegistry = TemporaryCommandRegistry.instance();
 
         // Register the 'Open Pre Req' command
-        context = await ExtensionUtil.registerPreReqAndReleaseNotesCommand(context);
-
+        context = await ExtensionUtil.registerPreActivationCommands(context);
+
+        if (extensionUpgraded) {
+            // If user upgraded major versions, if old local ansible environmens present ask if they should be teardown
+            // This needs to happen before rendering pannels, otherwise the user will see the old environments
+            const v1ExtDir: string = extDir.replace(`${path.sep}v2`, '');
+            FabricEnvironmentRegistry.instance().setRegistryPath(v1ExtDir);
+            try {
+                const v1EnvironmentEntries: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll();
```

Comment:
This returns ALL environments including environments which we've added with nodes, so they might be elsewhere.